src/components/frame/v5/pages/ActivityPage/hooks.tsx

```diff
@@ -1,25 +1,82 @@
 import clsx from 'clsx';
 import React from 'react';
+import { subDays, startOfDay } from 'date-fns';
 
+import { notNull } from '~utils/arrays';
+import {
+  useGetTotalColonyActionsQuery,
+  useGetTotalColonyDomainActionsQuery,
+} from '~gql';
 import { formatText } from '~utils/intl';
 import { WidthBoxItem } from '~v5/common/WidgetBoxList/types';
+import { useColonyContext } from '~hooks';
+import { createBaseActionFilter } from '~hooks/useActivityFeed/helpers';
+
+const getThirtyDaysAgoIso = () => {
+  const thirtyDaysAgo = subDays(new Date(), 30);
+  const midnightThirtyDaysAgo = startOfDay(thirtyDaysAgo);
+  return midnightThirtyDaysAgo.toISOString();
+};
 
 export const useActivityFeedWidgets = (): WidthBoxItem[] => {
+  const { colony } = useColonyContext();
+  const { domains, colonyAddress = '' } = colony ?? {};
+
+  const { data: totalActionData } = useGetTotalColonyActionsQuery({
+    variables: {
+      filter: {
+        ...createBaseActionFilter(colonyAddress),
+      },
+    },
+  });
+
+  const totalActions = totalActionData?.searchColonyActions?.total ?? 0;
+
+  const { data: recentActionData } = useGetTotalColonyActionsQuery({
+    variables: {
+      filter: {
+        ...createBaseActionFilter(colonyAddress),
+        createdAt: { gte: getThirtyDaysAgoIso() },
+      },
+    },
+  });
+
+  const recentActions = recentActionData?.searchColonyActions?.total ?? 0;
+
+  const { data: domainData } = useGetTotalColonyDomainActionsQuery({
+    variables: {
+      colonyId: colonyAddress,
+    },
+  });
+
+  const domainsActionCount =
+    // Buckets exists on this type but for some reason typescript is not
+    // recognising this
+    // @ts-ignore
+    domainData?.searchColonyActions?.aggregateItems[0]?.result?.buckets ?? [];
```

Comment:
I suspect it's the codegen that can't pick up union types without a separate plugin: https://the-guild.dev/graphql/codegen/plugins/other/fragment-matcher
It was pretty straightforward last time I set it up so I think it's worth getting sorted in this PR as we're gonna have the same problem with all GQL unions. Feel free to give me call if you need a hand with this

Reply:
turns out I wasn't understanding this issue clearly enough as the type was generated correctly have fixed this issue now :)